R/chunk_req_perform.R

```diff
@@ -0,0 +1,79 @@
+#' Perform a request in chunks
+#'
+#' @inheritParams req_perform
+#' @param chunk_size The size of each chunk.
+#' @param data The data to chunk.
+#' @param apply_chunk A function that applies the chunk to the request. It
+#'   takes two arguments:
+#'
+#'   1. `req`: the original request.
+#'   2. `chunk`: the current data chunk.
+#' @param parse_resp A function with one argument `resp` that parses the
+#'   response.
+#'
+#' @return A list of parsed responses.
+#' @export
+#'
+#' @examples
+#' ids <- 1:20
+#' responses <- request("https://api.restful-api.dev/objects") %>%
+#'   chunk_req_perform(
+#'     chunk_size = 10,
+#'     data = ids,
+#'     req_prep = function(req, chunk) {
+#'       chunk <- set_names(chunk, "id")
+#'       req_url_query(req, !!!chunk)
+#'     },
+#'     parse_resp = function(resp) {
+#'       # resp
+#'       parsed <- resp_body_json(resp)
+#'       data.frame(
+#'         id = sapply(parsed, function(x) x$id),
+#'         name = sapply(parsed, function(x) x$name)
+#'       )
+#'     }
+#'   )
+#'
+#' responses
+chunk_req_perform <- function(req,
```

Comment:
@hadley I think it might be better if we switch to the suffix form, i.e. `req_perform_chunk()`, `req_perform_paginate()` and `req_perform_multi()`. This makes it easier to discover by autocomplete.

Reply:
Yeah, I was thinking this as well.